src/classes/CON_ContactMerge_TDTM.cls

```diff
@@ -105,93 +112,253 @@ public class CON_ContactMerge_TDTM extends TDTM_Runnable {
     *   - recalculate Household Soft Credit OCRs 
     *   - recalculate Opportunity Rollups
     *
-    * @param listMergeKey The list of mergeKeys that provide contactIdLoser, accountIdLoser, and contactIdWinner
+    * @param mergeKeys The list of mergeKeys that provide contactIdLoser, accountIdLoser, and contactIdWinner
     * @return null
     ********************************************************************************************************/
-    @Future    
-    private static void handleContactMergeFixupsFuture(list<string> listMergeKey) {
-    
-        list<string> listContactIdLoser = new list<string>();
-        list<string> listAccountIdLoser = new list<string>();
-        list<string> listContactIdWinner = new list<string>();
+    @Future
+    private static void handleContactMergeFixupsFuture(List<String> mergeKeys) {
+        if (mergeKeys == null || mergeKeys.isEmpty()) {
+            return;
+        }
+
+        mergeProcessor.fixup(mergeKeys);
+    }
+
+    /*******************************************************************************************************
+    * @description Fixes records related to merged contacts
+    ********************************************************************************************************/
+    @TestVisible
+    private class ContactMergeProcessor {
+        @TestVisible
+        private List<String> losingContactIds;
```

Comment:
Even though `ContactMergeProcessor` is an inner class and the `fixup()` method called from the outer class, let's init collection vars when they are declared.